src/streams/send.rs

```diff
@@ -153,7 +153,7 @@ pub(crate) struct StreamTxStats {
     /// Can be null in case of missing or bad reports
     rtt: Option<f32>,
     /// losses collecter from RR (known packets, lost ratio)
-    losses: Vec<(u64, f32)>,
+    losses: Option<Vec<(u64, f32)>>,
```

Comment:
What I would suggest is, to make a newtype called `Losses` that has two constructors:

- `new`
- `no_op` / `dummy` etc

We then pass `Losses` as a parameter through all layers that don't yet know whether capturing `Losses` will be required. At the point where we have the config available, we can then decide which constructor to call.

This newtype can also encapsulate some of the functionality on how the data is actually processed and inserted, i.e. provide a more specific interface for the other components to interact with.

If you wanted to go full on type-safe, we could also define `Losses` as an enum such that in the no-op mode, there isn't even an underlying data-structure to operate on / fill.